conjurapi/client.go

```diff
@@ -667,13 +679,30 @@ func makeFullId(account, kind, id string) string {
 	return strings.Join(tokens, ":")
 }
 
-func parseID(fullID string) (account, kind, id string, err error) {
-	tokens := strings.SplitN(fullID, ":", 3)
-	if len(tokens) != 3 {
-		err = fmt.Errorf("Id '%s' must be fully qualified", fullID)
-		return
+// parseID accepts as argument a resource ID, and returns its components - account,
+// resource kind, and identifier. The provided ID can either be fully- or
+// partially-qualified. If the ID is fully-qualified, the account must match the
+// configured account. If the ID is partially-qualified, the configured account
+// will be used.
+//
+// Returns the resource's account, kind, identifier, and fully-qualified ID.
+//
+// Examples:
+// parseID("dev:user:alice") => "dev", "user", "alice", "dev:user:alice", nil
+// parseID("user:alice") => "dev", "user", "alice", "dev:user:alice", nil
+// parseID("prod:user:alice") => "", "", "", "", error
+func (c *Client) parseID(id string) (account, kind, identifier string, err error) {
+	tokens := strings.SplitN(id, ":", 3)
+	if len(tokens) == 3 {
+		if tokens[0] != c.config.Account {
+			return "", "", "", fmt.Errorf("Account of '%s' must match the configured account '%s'", id, c.config.Account)
+		}
```

Comment:
I think this might be too strict and we likely don't have to do this. Mainly because I'm still not 100% sure if it's not possible to make permission queries across accounts.